tensorflow/python/keras/layers/advanced_activations.py

```diff
@@ -333,3 +334,70 @@ def get_config(self):
   @tf_utils.shape_type_conversion
   def compute_output_shape(self, input_shape):
     return input_shape
+
+@keras_export('keras.layers.RReLU')
+class RReLU(Layer):
+  """Randomized Leaky Rectified Linear Unit function
+  It Randomizes the alpha passed to the leaky Relu function
+
+  'f(x) = alpha * x for x <0',
+  'f(x) = x for x>=0'.
+
+  where alpha is randomly sampled from uniform distribution U(lower,upper).
+
+  See: https://arxiv.org/pdf/1505.00853.pdf
+
+  Input shape:
+      Arbitrary. Use the keyword argument `input_shape`
+      (tuple of integers, does not include the samples axis)
+      when using this layer as the first layer in a model.
+
+  Output shape:
+      Same shape as the input.
+
+  Arguments:
+      lower: lower bound of the uniform distribution.
+      upper: upper bound of the uniform distribution.
+  """
+
+  def __init__(self, lower=0.125, upper=0.333, **kwargs):
+    super(RReLU, self).__init__(**kwargs)
+    if lower < 0 or upper < 0:
+      raise ValueError('lower/upper bound for RandomizedRectifierLayer needs '
+                       'to be > 0. ')
+    if lower > upper:
+      raise ValueError('Upper bound for RandomizedRectifierLayer needs '
+                       'to be higher than lower bound. ')
+
+    self.lower = lower
+    self.upper = upper
+    self.alpha = K.random_range(low=lower, high=upper)
```

Comment:
Without specifically using "K.cast_to_floatx", will this be able to support different floating point precision?

Reply:
At present all my TCs are passing , if you feel K.cast_to_floatx is better way to go i am up for change.